libs/indibase/abstractbaseclient.cpp

```diff
@@ -523,14 +517,9 @@ void AbstractBaseClient::sendNewNumber(INDI::Property pp)
 void AbstractBaseClient::sendNewNumber(const char *deviceName, const char *propertyName, const char *elementName,
                                        double value)
 {
-    INDI::BaseDevice *drv = getDevice(deviceName);
+    auto nvp = getDevice(deviceName).getNumber(propertyName);
```

Comment:
Such call does not result in crash is a device does not exist for deviceName?

Reply:
In case of non-existent `BaseDevice`, `BaseDevice` is returned marked with the method `isValid() == false`
Of course, `BaseDevice` can still be compared to nullptr and has additional operators to allow the use of init-statement in the condition.

### Example:
```cpp
#include <libindi/baseclient.h>
#include <libindi/basedevice.h>
#include <iostream>

int main(int, char *[])
{
    std::cout << "#1 property isValid: "
              << INDI::BaseClient().getDevice("invalid-device").getNumber("invalid-property").isValid()
              << std::endl;
    
    std::cout << "#2 isValid: "
              << INDI::BaseClient().getDevice("invalid-device").isValid()
              << std::endl;

    std::cout << "#3 compare device to nullptr: "
              << (INDI::BaseClient().getDevice("invalid-device") == nullptr)
              << std::endl;

    std::cout << "#4 compare device to not nullptr: "
              << (INDI::BaseClient().getDevice("invalid-device") != nullptr)
              << std::endl;

    if (auto device = INDI::BaseClient().getDevice("invalid-device"))
    {
        std::cout << "#5 device exists: " << device.getDeviceName() << std::endl;
    }
    else
    {
        std::cout << "#5 device not exists" << std::endl;
    }
    return 0;
}
```
### Output:
```
#1 property isValid: 0
#2 isValid: 0
#3 compare device to nullptr: 1
#4 compare device to not nullptr: 0
#5 device not exists
```